src/System.Json/tests/JsonArrayTests.cs

```diff
@@ -0,0 +1,337 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.IO;
+using System.Text;
+using Xunit;
+using System.Collections.Generic;
+using System.Collections;
+
+namespace System.Json.Tests
+{
+    public class JsonArrayTests
+    {
+        public static IEnumerable<object[]> JsonValues_TestData()
+        {
+            yield return new object[] { new JsonValue[0] };
+            yield return new object[] { new JsonValue[] { null } };
+        }
+
+        [Theory]
+        [MemberData(nameof(JsonValues_TestData))]
+        public void Ctor(JsonValue[] items)
+        {
+            VerifyJsonArray(new JsonArray(items), items);
+            VerifyJsonArray(new JsonArray((IEnumerable<JsonValue>)items), items);
+        }
+
+        private static void VerifyJsonArray(JsonArray array, JsonValue[] values)
+        {
+            Assert.Equal(values.Length, array.Count);
+            for (int i = 0; i < values.Length; i++)
+            {
+                Assert.Equal(values[i], array[i]);
+            }
+        }
+        
+        [Fact]
+        public void Ctor_Array_Works()
+        {
+            JsonValue[] items = new JsonValue[] { new JsonPrimitive(true) };
```

Comment:
I tried this. There was a problem. Xunit's `Assert.Equal(expected, actual)` internally performs more checks if `expected` or `actual` implement `ICollection`. `JsonValue` does in fact implement `ICollection`, *but* throws an `InvalidOperationException` all the time when `GetEnumerator` is called.

Furthermore, when Xunit resolves `[MemberData]` data, it calls `GetEnumerator` for some reason, I think related to a `string.Join` call, seen in the stack trace below:
![stacktrace](https://cloud.githubusercontent.com/assets/1275900/18842684/d6fa7c42-840d-11e6-825f-fdd377335d70.png)

Unfortunately this means that we can't put these tests in `MemberData`, so this is a workaround